Netflix/metaflow
# [Ready for Review] Improve native resume

Extend the resume speedup change to native(local) resume.

Note:
* We will go over all the successful tasks and copy them first. Then we reconstruct runtime queue to continue.
* We will keep UBF resume behavior the same. i.e if some mapper tasks fail, all mapper tasks will reran in the resume.
     
A few test example flows:
* Resume w/ some branch: [link](https://gist.github.com/darinyu/b895a8aca64f4674a147475763a50a85)
* Resume during failure on foreach split :[link](https://gist.github.com/darinyu/5ca42e7bad8e984f6bff942e1f064ff4)
* Resume during failure on UBF split: [link](https://gist.github.com/darinyu/aaf604a9e1e22c51c958135974caa27a)
* Resume during failure on UBF join: [link](https://gist.github.com/darinyu/1393a506b77b874c0cffeadaac2bfb5d)

To test the above flows:
```python
python flow.py run
python flow.py resume
```

Benchmark:
![Resume-Speed-Test-Google-Docs](https://github.com/Netflix/metaflow/assets/5545942/f04cf62c-0f93-4aae-8670-1b3ca3184b8a)


Open question:
* We only consider last "run" but never consider last "resume", so running multiple resume will anchor on the same run_id (instead of previous resume). If running resume consecutively, should we continue on the resume?
